src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -0,0 +1,235 @@
+        thread[0].start()
+
+        # 4.
+        self.log_step(
+            "Marking the {} device as faulty and verifying it is 'EVICTED' and its "
+            "LED is 'ON'".format(test_dev))
+        if not self.check_result(self.set_device_faulty(test_dev), "EVICTED", "ON"):
+            self.fail("#Result of set_device_fault, device not in expected EVICTED, ON state")
+        # check device state after set nvme-faulty
+        time.sleep(1)
+        if not self.verify_dev_led_state(test_dev, "NORMAL", "ON"):
+            self.fail("#After set_device_fault, device not back to NORMAL, ON state")
+
+        # 5.
+        self.log_step("Waiting for IOR to complete")
+
+        # 6.
+        thread[0].join()
+        self.log_step("IOR completed and check for error")
+        errors = 0
+        while not thread_queue.empty():
+            result = thread_queue.get()
+            self.log.debug("Results from thread %s (log %s)", result["job_id"],
+                           result["log"])
+            for name in ("command", "exit_status", "interrupted", "duration"):
+                self.log.debug("  %s: %s", name, getattr(result["result"], name))
+            for name in ("stdout", "stderr"):
+                self.log.debug("  %s:", name)
+                for line in getattr(result["result"], name).splitlines():
+                    self.log.debug("    %s:", line)
+            if result["result"].exit_status != 0:
+                errors += 1
+        if errors:
+            self.fail("Errors running IOR {} thread".format(errors))
+
+        # 7.
+        self.log_step("Check drive status as 'NORMAL' and led status is 'ON'")
+        if not self.verify_dev_led_state(test_dev, "NORMAL", "ON"):
+            self.fail(
+                "#After set_device_fault, IOR completed, device not in expected NORMAL, ON state")
+
+        # Verify IOR data and check container
+        self.log.info("Verify IOR data")
+        ior_kwargs["ior_params"]["flags"] = self.ior_r_flags
+        ior_kwargs["log"] = "ior_read_pool_test.log"
+        try:
+            run_ior(**ior_kwargs)
+        except CommandFailure as error:
+            self.fail("Error in ior read {}.".format(error))
+        kwargs = {"pool": self.pool.identifier,
+                  "cont": self.container.identifier}
+        output = self.daos_command.container_check(**kwargs)
+        self.log.info(output)
+
+        # 8.
+        self.log_step("Replace the same drive back.")
+        result = self.dmg.storage_replace_nvme(old_uuid=test_dev, new_uuid=test_dev)
+
+        # 9.
+        self.log_step("Drive status LED should be off indicating good device is plugged-in.")
+        if not self.check_result(result, "NORMAL", "OFF"):
+            self.fail("#After storage replace nvme, device not in expected NORMAL, OFF state")
+
+        # 10.
+        self.log_step("Reset and cleanup")
+        for device in devices:
+            self.run_vmd_led_identify(device, reset=True)
```

Comment:
Can be removed with `register_cleanup` implementation.
```suggestion
        self.log.info("Test passed")
```